infra/iam.tf

```diff
@@ -55,7 +55,7 @@ module "iam_policy"{
                 "ecr:GetDownloadUrlForLayer",
                 "ecr:PutImage"
             ],
-            "Resource": "${module.ecr.repository_arn}"
+          "Resource": "*"

```

Comment:
Check docs so you don't have to use wildcard in order to have principle of least privilege.

https://docs.aws.amazon.com/AmazonECR/latest/userguide/image-push-iam.html